Web/Edubase.Web.UI/Areas/Groups/Controllers/GroupController.cs

```diff
@@ -845,23 +845,25 @@ private async Task<GroupEditorViewModel> PopulateSelectLists(GroupEditorViewMode
 
         private GroupEditorViewModel SetEditPermissions(GroupEditorViewModel viewModel)
         {
-            viewModel.CanUserCloseAndMarkAsCreatedInError = viewModel.GroupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust, GT.SchoolSponsor, GT.Federation)
+            //this method only controls the visibility of the fields not access to the actual page
+
+            viewModel.CanUserCloseAndMarkAsCreatedInError = viewModel.GroupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust, GT.SchoolSponsor, GT.Federation, GT.SecureSingleAcademyTrust)
                                                                && !viewModel.StatusId.OneOfThese(GS.CreatedInError, GS.Closed)
-                                                               && User.InRole(AuthorizedRoles.IsAdmin);
+                                                               && User.InRole(AuthorizedRoles.CanUserCloseAndMarkAsCreatedInError);
 
```

Comment:
I think we would have to rewrite the way that it works to look at the establishment type to fix this.  I agree it would be better  if it said if the establishment type is x and the group is y they can... But I have really just added to the bad way it works now.

Reply:
The logic before you added YCS is that backoffice (Admin) user can CloseAndMarkAsCreatedInError all of the listed group types (MAT, SAT, SchoolSponsor, Federation), which I'm assuming they can. By adding YCS, you're saying that YCS can also have the same rights for all of those group types, which isn't correct.
You could just split the logic into two? - keep the first bit as was and add an OR for the YCS user
viewModel.CloseAndMarkAsCreatedInError = (rules for backoffice user) || (rules for YCS user)
or you could 'go the whole hog' and extract all the permissions logic out into a separate class.